test/e2e/on-request-error/server-action-error/server-action-error.test.ts

```diff
@@ -0,0 +1,145 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry } from 'next-test-utils'
+
+describe('on-request-error - server-action-error', () => {
+  const { next, skipped } = nextTestSetup({
+    files: __dirname,
+    skipDeployment: true,
+    env: {
+      __NEXT_EXPERIMENTAL_INSTRUMENTATION: '1',
+    },
+  })
+
+  if (skipped) {
+    return
+  }
+
+  const outputLogPath = 'output-log.json'
+
+  async function getOutputLogJson() {
+    if (!(await next.hasFile(outputLogPath))) {
+      return {}
+    }
+    const content = await next.readFile(outputLogPath)
+    return JSON.parse(content)
+  }
+
+  async function validateErrorRecord({
+    name,
+    url,
+    renderSource,
+    renderType,
+    isMiddleware = false,
+  }: {
+    name: string
+    url: string
+    renderSource: string | undefined
+    renderType: string | undefined
+    isMiddleware?: boolean
+  }) {
+    await retry(async () => {
+      const recordLogs = next.cliOutput
+        .split('\n')
+        .filter((log) => log.includes('[instrumentation] write-log'))
+      const expectedLog = recordLogs.find((log) => log.includes(name))
+      expect(expectedLog).toBeDefined()
+    }, 5000)
+
+    const json = await getOutputLogJson()
+    const record = json[name]
+
+    expect(record).toBeDefined()
+    const { payload, count } = record
+    expect(payload.message).toBe(name)
+    expect(count).toBe(1)
```

Comment:
```suggestion
    expect(record).toEqual(expect.objectContaining({ message, count: 1 }))
```


The more assertions, the less readable the error will be. Once one of these fails, you probably want to see the full object you received since that may contain important debug information